website/docs/final.md

```diff
@@ -0,0 +1,98 @@
+---
+id: final
+title: Final Methods, Classes, and Modules
+---
+
+Sorbet supports final methods, classes, and modules.
+
+> **Note**: Some of the support for final is only implemented using the runtime
+> system. Support for these checks in the static system is planned for the
+> future.
+
+## Definitions
+
+- A final method `m` on a class or module `A` is a method such that
+  - any class or module that includes, extends, or inherits `A` may not define
+    its own method `m`
+  - `A` itself may only define `m` exactly once
+- A final class or module `F` is a class or module such that
+  - no class or module may include, extend, or inherit `F`
+  - every method defined on `F` is final
+
+## Enabling all of the runtime checks
+
+Before using final in a project, we recommend opting in to all of the runtime
+checks.
```

Comment:
Not sure what you mean by "is it possible to forget?" The hooks on Module, Class, etc are opt-in so they're off by default. In that sense, I suppose someone could intend to turn these hooks on but forget to call the special method.

Reply:
I guess, can we make `sig(:final)` raise if the hook isn't there?

Reply:
I guess I'm looking for a clarification on why it's a recommendation not a requirement.

Reply:
It's a recommendation because there are some final checks that run even if this method is not called. Should we make it so that you must call this method in order to use any final features?

Reply:
Ah that was what I didn't understand. Could you make that more clear in the doc?